File: component/db/Class_db.pas
unit Class_db;

interface

uses
  mysql.data.mysqlclient,
  system.configuration,
  system.data;

type
  TClass_db = class abstract
  private
    { Private Declarations }
  protected
    connection: mysqlconnection;
    procedure Close;
    procedure Open;
  public
    constructor Create;
  end;

implementation

constructor TClass_db.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  connection := mysqlconnection.Create;
  connection.ConnectionString := configurationmanager.AppSettings['db_connection_string'];
end;

procedure TClass_db.Close;
begin
  connection.Close;
end;

procedure TClass_db.Open;
begin
  if connection.State <> connectionstate.OPEN then begin
    connection.Open;
  end;
end;

end.

File: component/db/Class_db.cs
using MySql.Data.MySqlClient;
using System.Configuration;
using System.Data;

namespace Class_db
{
    public abstract class TClass_db
    {
        protected MySqlConnection connection = null;
        //Constructor  Create()
        public TClass_db() : base()
        {
            // TODO: Add any constructor code here
            connection = new MySqlConnection();
            connection.ConnectionString = ConfigurationManager.AppSettings["db_connection_string"];
        }
        protected void Close()
        {
            connection.Close();
        }

        protected void Open()
        {
            if (connection.State != ConnectionState.Open)
            {
                connection.Open();
            }
        }

    } // end TClass_db

}
